### ISSUE TITLE
Delete of soft deleted rows fails

### ISSUE THREAD

**Issue by Jerry-Carter** [vapor/fluent#627]
If you have soft delete enabled, deleting a row will set the column associated with the 'deletedAtKey'.  Following this with a forced delete will not result in the row being deleted.  The following SQL is generated on PostgreSQL:

DELETE FROM "MyTable" WHERE ("MyTable"."deletedAt" IS NULL OR "MyTable"."deletedAt" > ($1)) AND "MyTable"."id" = ($2) [0x00022aab9bdf99ee (TIMESTAMP), 24 (BIGINT)]

where the timestamp ($1) is after the deletedAtKey column value.  The fix may be as simple as changing '>' to '<'.  There is currently not a unit test for this case.

**Comment by grundoon** [comment]
Seems to me that there should be no `deletedAt` check at all if `delete(force: true, on: ...)`

**Comment by Jerry-Carter** [comment]
I'm extending BenchmarkSoftDeletable.swift to cover this case.  Will commit once I'm able to run it.

**Comment by MrMage** [comment]
I am using the following, which should work:

```
User.query(on: connection, withSoftDeleted: true)
					.filter(\User.id == userID)
					.delete(force: true)
```

**Comment by Jerry-Carter** [comment]
Pull request: https://github.com/vapor/fluent/pull/628

**Comment by tanner0101** [comment]
Fixed in #628
